python/ray/data/_internal/execution/operators/base_physical_operator.py

```diff
@@ -12,6 +13,13 @@
 from ray.data.context import DataContext
 
 
+class InternalQueueOperatorMixin(PhysicalOperator, abc.ABC):
+    @abc.abstractmethod
+    def internal_queue_size(self) -> int:
+        """Returns Operator's internal queue size"""
+        ...
```

Comment:
what's the benefit of introducing this Mixin?  
I feel this makes the dependency more convoluted. 
If we want to force subclasses to implement this. We can just make it abstract in the PhysicalOperator.

Reply:
It's a typical mixin use-case -- to abstract and inject a scope of functionality into the class

 - We don't want to force every class to implement it (it's only relevant for a handful of classes)
 - This mixin will be expanded with the queue itself offloaded to it (at later point to avoid overloading this PR)

Reply:
> This mixin will be expanded with the queue itself offloaded to it (at later point to avoid overloading this PR)

This is a good point. Let's leave a TODO here?